power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -96,22 +101,131 @@ inline void update_inverse(MainModelState<ComponentContainer> const& state, Forw
+    (void)y_bus_param; // suppress compiler unused variable warning
+    assert(y_bus_param.branch_param.size() == math_model_param->branch_param.size());
+    assert(y_bus_param.shunt_param.size() == math_model_param->shunt_param.size());
+    assert(y_bus_param.source_param.size() == math_model_param->source_param.size());
+
+    if (seq_idx_map == nullptr) {
+        return;
+    }
+
+    // when sequence_idx_map available: loop through all branches and shunts of math_model_param, check changes
+    auto querry_param_in_seq_map = [&seq_idx_map](IdxVector const& param_idd_vec) {
+        std::vector<Idx2D> result;
+        for (auto const& param_idd : param_idd_vec) {
+            auto const& seq_idx = (*seq_idx_map)[param_idd];
+            result.insert(result.end(), seq_idx.begin(), seq_idx.end());
+        }
+        return result;
+    };
+
+    auto branch_params = querry_param_in_seq_map(y_bus.get_branch_param_idx());
+    auto shunt_params = querry_param_in_seq_map(y_bus.get_shunt_param_idx());
+
+    MathModelParamIncrement<sym> math_model_param_incrmt;
+    math_model_param_incrmt.branch_param_to_change.reserve(branch_params.size());
+    std::transform(branch_params.begin(), branch_params.end(),
+                   std::back_inserter(math_model_param_incrmt.branch_param_to_change),
+                   [](const Idx2D& val) { return val.pos; });
+    math_model_param_incrmt.shunt_param_to_change.reserve(shunt_params.size());
+    std::transform(shunt_params.begin(), shunt_params.end(),
+                   std::back_inserter(math_model_param_incrmt.shunt_param_to_change),
+                   [](const Idx2D& val) { return val.pos; });
+
+    // check changes in case sequence_idx_map is available but indicates empty
+    // this back-up check fixes the edge case where the sequence_idx_map is available but indicates empty
+    if (branch_params.size() == 0) {
+        auto branch_param_to_change_views =
+            boost::irange(y_bus_param.branch_param.size()) |
+            boost::adaptors::filtered([&math_model_param, &y_bus_param](Idx i) {
+                return cmplx_neq<sym>(math_model_param->branch_param[i].yff(), y_bus_param.branch_param[i].yff()) ||
+                       cmplx_neq<sym>(math_model_param->branch_param[i].yft(), y_bus_param.branch_param[i].yft()) ||
+                       cmplx_neq<sym>(math_model_param->branch_param[i].ytf(), y_bus_param.branch_param[i].ytf()) ||
+                       cmplx_neq<sym>(math_model_param->branch_param[i].ytt(), y_bus_param.branch_param[i].ytt());
+            });
+        math_model_param_incrmt.branch_param_to_change.reserve(
+            std::distance(branch_param_to_change_views.begin(), branch_param_to_change_views.end()));
+        math_model_param_incrmt.branch_param_to_change = {branch_param_to_change_views.begin(),
+                                                          branch_param_to_change_views.end()};
```

Comment:
Since we are using `boost::filtered_ranges` here, reserve with a `size()` call is not possible. Reserve can of course be removed, simply added for potential performance benefit.

Reply:
Also `std::move` requires more work due to the same cause.

Reply:
Resolved in latest commit.

Reply:
Resolved in latest commit.